app/models/follow_recommendation_filter.rb

```diff
@@ -11,10 +11,19 @@ class FollowRecommendationFilter
   attr_reader :params, :language
 
   def initialize(params)
-    @language = params.delete('language') || I18n.locale
+    @language = usable_language(params.delete('language') || I18n.locale)
     @params   = params
   end
 
+  def usable_language(locale)
+    return locale if Trends.available_locales.include?(locale)
+
+    locale = locale.to_s.split(/[_-]/).first
+    return locale if Trends.available_locales.include?(locale)
+
+    nil
+  end
```

Comment:
How about making this method private and moving it to the bottom? It seems like a small utility method that should not be called from outside.